src/state/megaStore.tsx

```diff
@@ -74,6 +76,12 @@ export const Provider: ParentComponent = (props) => {
         deleting: false,
         scan_result: undefined as ParsedParams | undefined,
         price: 0,
+        fiat: {
```

Comment:
I think you could do loading like this so you're not parsing every time in sync? just seems noisy code-wise:

```
        fiat: localStorage.getItem("currency") ? JSON.parse(localStorage.getItem("currency")!) as Currency : {
            value: "USD",
            label: "United States Dollar USD",
            hasSymbol: "$",
            maxFractionalDigits: 2
        },
```

then in sync you can use state.fiat knowing that it exists...

not a huge deal, I'm mostly just trying not to have the sync function overwhelmed by price-specific logic